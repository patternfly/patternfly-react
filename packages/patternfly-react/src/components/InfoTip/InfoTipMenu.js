import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

// This must be a class component react-bootstrap passes a ref to it.
// eslint-disable-next-line react/prefer-stateless-function
class InfoTipMenu extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node.isRequired,
    bsRole: PropTypes.string,
    rootCloseEvent: PropTypes.oneOf(['click', 'mousedown'])
  };

  static defaultProps = {
    bsRole: 'menu',
    className: '',
    rootCloseEvent: 'click'
  };

  render() {
    const {
      children,
      className,
      bsRole,
      rootCloseEvent,
      labelledBy, // eslint-disable-line react/prop-types
      pullRight, // eslint-disable-line react/prop-types
      bsClass, // eslint-disable-line react/prop-types
      ...props
    } = this.props;
    const infoTipMenuClass = classNames('dropdown-menu', 'infotip', 'bottom-right', className);

    return (
      <div className={infoTipMenuClass} style={{ padding: '' }} {...props}>
        <div className="arrow" />
        {children}
      </div>
    );
  }
}

export default InfoTipMenu;
