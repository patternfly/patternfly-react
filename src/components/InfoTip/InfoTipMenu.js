import React from 'react';
import ClassNames from 'classnames';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class InfoTipMenu extends React.Component {
  render() {
    const {
      children,
      className,
      bsRole,
      rootCloseEvent,
      labelledBy, // eslint-disable-line react/prop-types
      pullRight, // eslint-disable-line react/prop-types
      bsClass, // eslint-disable-line react/prop-types
      ...props
    } = this.props;

    const infoTipMenuClass = ClassNames(
      'dropdown-menu',
      'infotip',
      'bottom-right',
      className
    );
    return (
      <div className={infoTipMenuClass} style={{ padding: '' }} {...props}>
        <div className="arrow" />
        {children}
      </div>
    );
  }
}

InfoTipMenu.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  bsRole: PropTypes.string,
  rootCloseEvent: PropTypes.oneOf(['click', 'mousedown'])
};
InfoTipMenu.defaultProps = {
  bsRole: 'menu',
  className: '',
  rootCloseEvent: 'click'
};
export default InfoTipMenu;
