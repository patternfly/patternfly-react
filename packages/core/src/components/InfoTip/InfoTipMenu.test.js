import React from 'react';
import { shallow } from 'enzyme';
import InfoTipMenu from './InfoTipMenu';

const props = {
  children: 'children'
};

test('forwards only valid props to the root element', () => {
  const view = shallow(
    <InfoTipMenu
      {...props}
      id="id"
      bsRole="bsRole"
      bsClass="bsClass"
      pullRight="pullRight"
      labelledBy="labeledBy"
    />
  );
  expect(view).toMatchSnapshot('props are forwarded to root element');
});

test('adds classname to the root element', () => {
  const className = 'extra-classname';
  const view = shallow(<InfoTipMenu {...props} className={className} />);
  const rootClassName = view.first().props().className;
  expect(rootClassName.includes(className)).toBe(true);
});
