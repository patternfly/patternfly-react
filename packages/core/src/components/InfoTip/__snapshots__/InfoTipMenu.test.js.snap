// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forwards only valid props to the root element: props are forwarded to root element 1`] = `
<div
  className="dropdown-menu infotip bottom-right"
  id="id"
  style={
    Object {
      "padding": "",
    }
  }
>
  <div
    className="arrow"
  />
  children
</div>
`;
